assets/src/utils/parse-html-placeholders.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * External imports
+ */
+import { isEmpty } from 'lodash';
+
+const PLACEHOLDER = '%%var%%';
+
+/**
+ * kinda like a weaker version of sprintf that won't choke on HTML elements
+ *
+ * @function
+ * @param {string} placeholderText
+ * @param {Array} replacements
+ * @return {string} text with placeholders replaced by variables
+ */
+const parseHtmlPlaceholders = ( placeholderText, replacements ) => {
+	if ( isEmpty( replacements ) ) {
+		return placeholderText;
+	}
+	if ( ! Array.isArray( replacements ) ) {
+		replacements = [ replacements ];
+	}
+	if ( ! placeholderText.includes( PLACEHOLDER ) ) {
+		throw new RangeError(
+			'The provided string does not have any placeholders.'
+		);
+	}
+	const finalText = [];
+	const chunks = placeholderText.split( PLACEHOLDER );
+	if ( chunks.length - replacements.length !== 1 ) {
```

Comment:
Similarly here, if the number of replacements is more than the number of placeholders, it shouldn't be a problem I think. So, it could be
```JS
if ( chunks.length > replacements.length ) {
	...
}
```

Reply:
my thinking was that the number of chunks would be greater than the number of replacements by **one**, ie:

```js
const replacements = [ 'placeholder' ];
const placeholderText = `this text has a %%var%% in it`;
const chunks = placeholderText.split( PLACEHOLDER );
console.log( 'chunks: ', chunks.length );
// chunks: 2
console.log( 'replacements : ', replacements .length );
// replacements : 1
```

...but that may not always be case.

will need to look into this more to ensure that there is always a consistent ration of chunks to replacements, else find a better way to handle this

Reply:
A strict check isn't a problem though.

Reply:
kk just confirmed this:

```js
const placeholderText = `sample text%%var%%`;
const chunks = placeholderText.split( '%%var%%' );
console.log( 'chunks: ', chunks.length );
// chunks:  2 (2) ["sample text", ""]

const placeholderText2 = `%%var%%sample text%%var%%`;
const chunks2 = placeholderText2.split( '%%var%%' );
console.log( 'chunks: ', chunks2.length, chunks2 );
// chunks:  3 (3) ["", "sample text", ""]
```

so the number of chunks should always be **one** more than the number of replacements, so I think my original condition:

```js
if ( chunks.length - replacements.length !== 1 ) {
```

should remain as is imho and simply ensuring that there are more chunks than replacements, like:

```js
if ( chunks.length > replacements.length ) {
```

does not protect against something like:

```js
const replacements = [ 'only one placeholder' ];
const placeholderText = `%%var%%sample text%%var%%`;
```

which would error when we attempt to access the replacement for the second placeholder.

That said, I'm still open to other suggestions for improving this, but please note that i didn't intend on having the logic smart enough to determine whether it should reuse the same replacement for multiple placeholders.

Reply:
Great! Thanks